File: android/app/src/main/java/com/pulseos/data/api/HomeService.kt
package com.pulseos.data.api

import com.google.gson.reflect.TypeToken
import com.pulseos.core.network.ApiClient
import com.pulseos.data.dto.DashboardDTO

class HomeService(private val client: ApiClient) {

    private val dashboardType = object : TypeToken<DashboardDTO>() {}

    suspend fun getDashboard(): DashboardDTO =
        client.get("/api/v1/home/dashboard", dashboardType)
}

File: ios/PulseOS/Data/API/HomeService.swift
import Foundation

actor HomeService {
    private let client: APIClient

    init(client: APIClient) {
        self.client = client
    }

    func getDashboard() async throws -> DashboardDTO {
        try await client.get("/api/v1/home/dashboard", as: DashboardDTO.self)
    }
}
